substrate/client/rpc-spec-v2/src/transaction/transaction_broadcast.rs

```diff
@@ -0,0 +1,292 @@
+
+			while !is_done {
+				// Wait for the last block to become available.
+				let Some(best_block_hash) =
+					last_stream_element(&mut best_block_import_stream).await
+				else {
+					return;
+				};
+
+				let mut stream = match pool
+					.submit_and_watch(best_block_hash, TX_SOURCE, decoded_extrinsic.clone())
+					.await
+				{
+					Ok(stream) => stream,
+					// The transaction was not included to the pool.
+					Err(e) => {
+						let Ok(pool_err) = e.into_pool_error() else { return };
+
+						if is_pool_error_recoverable(&pool_err) {
+							// Try to resubmit the transaction at a later block for recoverable
+							// errors.
+							continue;
+						} else {
+							return;
+						}
+					},
+				};
+
+				while let Some(event) = stream.next().await {
+					match event {
+						// The transaction propagation stops when:
+						// - The transaction was included in a finalized block via
+						//   `TransactionStatus::Finalized`.
+						TransactionStatus::Finalized(_) |
+						// - The block in which the transaction was included could not be finalized for
+						//   more than 256 blocks via `TransactionStatus::FinalityTimeout`. This could
+						//   happen when:
+						//   - the finality gadget is lagging behing
+						//   - the finality gadget is not available for the chain
+						TransactionStatus::FinalityTimeout(_) |
+						// - The transaction has been replaced by another transaction with identical tags
+						// (same sender and same account nonce).
+						TransactionStatus::Usurped(_) => {
+							is_done = true;
+							break;
+						},
+
+						// Dropped transaction may enter the pool at a later time, when other
+						// transactions have been finalized and remove from the pool.
+						TransactionStatus::Dropped |
+						// An invalid transaction may become valid at a later time.
+						TransactionStatus::Invalid => {
+							break;
+						},
+
+						// The transaction is still in the pool, the ready or future queue.
+						TransactionStatus::Ready | TransactionStatus::Future |
+						// Transaction has been broadcasted as intended.
+						TransactionStatus::Broadcast(_) |
+						// Transaction has been included in a block, but the block is not finalized yet.
+						TransactionStatus::InBlock(_) |
+						// Transaction has been retracted, but it may be included in a block at a later time.
+						TransactionStatus::Retracted(_) => (),
+					}
+				}
+			}
+		};
+
+		// Convert the future into an abortable future, for easily terminating it from the
+		// `transaction_stop` method.
+		let (fut, handle) = futures::future::abortable(broadcast_transaction_fut);
+		// The future expected by the executor must be `Future<Output = ()>` instead of
+		// `Future<Output = Result<(), Aborted>>`.
+		let fut = fut.map(drop);
+
+		// Keep track of this entry and the abortable handle.
+		{
+			let mut broadcast_ids = self.broadcast_ids.write();
+			broadcast_ids.insert(id.clone(), BroadcastState { handle });
```

Comment:
Good catch! I've removed the `operationId` from our internal mappings as soon as the broadcast future exits.

I believe this is in-line with the spec; since the [tx_stop](https://github.com/paritytech/json-rpc-interface-spec/blob/main/src/api/transaction_unstable_stop.md) mentions it works for active broadcasts:

> A JSON-RPC error is generated if the operationId doesn't correspond to any active transaction_unstable_broadcast operation.

As a side-effect have added a small task executor wrapper for our testing; just to make sure we don't race between tx_stop and broadcast future clearing the internal state